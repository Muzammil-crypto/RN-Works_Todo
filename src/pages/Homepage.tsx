import React, {useContext} from 'react';
import {Text, View, TextInput, Button, ScrollView} from 'react-native';
import {TodoContext} from '../contexts/TodoContext';
import {styles} from '../styles/Todo.component.styles';
export interface IToDo {
  text: string;
  completed: boolean;
}

export const Homepage = () => {
  const {value, setValue, error, setError, toDoList, setTodoList} =
    useContext(TodoContext);
  //Functoins
  const onChangeText = (e: any): void => {
    setValue!(e);
    setError!(false);
  };
  const handleSubmit = (): void => {
    if (value.trim()) {
      setTodoList!([...toDoList, {text: value, completed: false}]);
    } else {
      setError!(true);
    }
    setValue!('');
  };

  const removeItem = (index: number): void => {
    const newToDoList = [...toDoList];
    newToDoList.splice(index, 1);
    setTodoList!(newToDoList);
  };

  const toggleComplete = (index: number): void => {
    const newToDoList = [...toDoList];
    newToDoList[index].completed = !newToDoList[index].completed;
    setTodoList!(newToDoList);
  };

  return (
    <ScrollView>
      <View style={styles.container}>
        <Text style={styles.title}>Create a Task </Text>
        <Text style={styles.subtitle1}>Enter the Task You Want To Create</Text>
        <View style={styles.inputContainer}>
          <TextInput
            placeholder="Enter your todo task..."
            value={value}
            onChangeText={e => onChangeText(e)}
            style={styles.inputBox}
          />
          {error && <Text style={styles.error}> Input field is empty...</Text>}
          <Button title="Add Task" onPress={handleSubmit} />
        </View>

        <Text style={styles.subtitle}>Your Tasks :</Text>
        {toDoList.length === 0 && <Text>No to do task available :(</Text>}
        {toDoList.map((toDo: IToDo, index: number) => (
          <View style={styles.listItem} key={`${index}_${toDo.text}`}>
            <Text
              style={[
                styles.task,
                // eslint-disable-next-line react-native/no-inline-styles
                {textDecorationLine: toDo.completed ? 'line-through' : 'none'},
              ]}>
              {toDo.text}
            </Text>
            <Button
              title={toDo.completed ? 'Completed' : 'Complete'}
              onPress={() => toggleComplete(index)}
            />
            <Button
              title="X"
              onPress={() => {
                removeItem(index);
              }}
              color="crimson"
            />
          </View>
        ))}
      </View>
    </ScrollView>
  );
};
