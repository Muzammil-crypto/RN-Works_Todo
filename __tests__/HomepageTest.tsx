import React from 'react';
import renderer from 'react-test-renderer';
import {Homepage} from '../src/pages/Homepage';

test('renders correctly', () => {
  const tree = renderer.create(<Homepage />).toJSON();
  expect(tree).toMatchSnapshot();
});
