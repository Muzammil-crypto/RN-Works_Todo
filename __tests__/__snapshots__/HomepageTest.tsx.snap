// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "height": 1334,
          "padding": 20,
        }
      }
    >
      <Text
        style={
          {
            "color": "#009243",
            "fontSize": 30,
            "fontWeight": "bold",
            "marginBottom": 5,
            "marginTop": 80,
          }
        }
      >
        Create a Task 
      </Text>
      <Text
        style={
          {
            "color": "black",
            "fontSize": 14,
            "marginBottom": 20,
          }
        }
      >
        Enter the Task You Want To Create
      </Text>
      <View
        style={
          {
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginBottom": 20,
            "width": "100%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your todo task..."
          style={
            {
              "borderColor": "#009243",
              "borderRadius": 8,
              "borderWidth": 0.5,
              "marginBottom": 10,
              "padding": 8,
              "width": "100%",
            }
          }
          value=""
        />
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              [
                {},
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Add Task
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={
          {
            "color": "#009243",
            "fontSize": 20,
            "marginBottom": 20,
          }
        }
      >
        Your Tasks :
      </Text>
      <Text>
        No to do task available :(
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
